### PR TITLE
[WEBDEV-492] Restructuring of tracks component

### PR DESCRIPTION
**Pull request by vinternet-pds**
* Tracks component refactored into correct list item structure
* Track margins and nested heading margins refactored
* Options added for track headings to have typography weight variants

### CODE REVIEW

File: src/stylesheets/elements/_headings.scss [hunk 1]
```diff
@@ -153,30 +153,49 @@ footer h2 {
...
-  h2 + h2, h3 + h3, h4 + h4, h5 + h5 {
-    margin-top: $base-spacing-unit-large;
+  li:first-child > {
+    h2, h3, h4, h5, h6 {
+      margin-top: 0;
+    }
   }
 }
+
+.track__light {
```
Comment (joshuawaheed) [inline src/stylesheets/elements/_headings.scss]: Can you rename this to `.track--light`? Then we can add it as an extra class which makes it light. Then it can be similar to the way we are handling lists. For example, when we handle a 3 columned block-list, we would mark it up as `.list--block.list--3`.

File: src/stylesheets/elements/_headings.scss [hunk 2]
```diff
@@ -193,6 +212,13 @@ footer h2 {
   }
 }
 
+.track--alternative__light {
```
Comment (joshuawaheed) [inline src/stylesheets/elements/_headings.scss]: By changing `.track__light` to `.track--light`, we won't need this anymore. We can instead use them together in our HTML like this,

```
<ul class="track--alternative track--light">
  ...
</ul>
```

File: src/stylesheets/elements/_headings.scss [hunk 3]
```diff
@@ -153,30 +153,49 @@ footer h2 {
...
+.track__light {
+  @extend .track;
+  li > {
+    h2, h3, h4, h5, h6 {
+      @include typography--light;
+    }
+  }
+}
+
 .track--alternative {
```
Comment (joshuawaheed) [inline src/stylesheets/elements/_headings.scss]: If possible, can we move this to *_track.scss* as it doesn't look related to headings?

File: src/stylesheets/elements/_headings.scss (hunk 1 above)
Comment (vinternet-pds) [reply src/stylesheets/elements/_headings.scss]: As discussed, we'll have a collective chat about this - if a new track--light component is needed I'll create later. I have removed track--light from code base for now.

File: src/stylesheets/elements/_headings.scss (hunk 3 above)
Comment (vinternet-pds) [reply src/stylesheets/elements/_headings.scss]: Done - make sense

File: src/stylesheets/elements/_headings.scss [hunk 4]
```diff
@@ -153,30 +153,49 @@ footer h2 {
...
   @extend .track;
   &:before {
```
[the remark this answers is not in the document]
Comment (vinternet-pds) [reply src/stylesheets/elements/_headings.scss]: Great improvement! code updated
